### PR TITLE
add files for review

### PR DESCRIPTION
**Pull request by 27149chen**
files for review:
1. README.md
2. ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml

### CODE REVIEW

File: README.md-review [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Operator for IBM block storage CSI driver
+The Container Storage Interface (CSI) Driver for IBM block storage systems enables container orchestrators such as Kubernetes to manage the life-cycle of persistent storage.
```
Comment (rivkap) [inline README.md-review]: "life cycle" (no hyphen)

File: README.md-review [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+This is the official operator to deploy and manage IBM block storage CSI driver.
+
+Supported container platforms:
+  - Openshift v4.1
```
Comment (rivkap) [inline README.md-review]: OpenShift

File: README.md-review [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  - Kubernetes v1.13
+
+Supported operating systems:
+  - RHEL 7.x (x86 architecture)
+  - RHCOS 4.1 (x86 architecture)
+
+## Prerequisite
```
Comment (rivkap) [inline README.md-review]: Prerequisites

File: README.md-review [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+Please go to section [`Prerequisites for Driver Installation`](https://github.com/IBM/ibm-block-csi-driver#prerequisites-for-driver-installation) for details.
```
Comment (rivkap) [inline README.md-review]: Please see [,.....] for details.

File: README.md-review [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+## Installation
+
+### Install the operator
+1. Download the manifest from GitHub.
+```bash
+curl https://raw.githubusercontent.com/IBM/ibm-block-csi-operator/develop/deploy/ibm-block-csi-operator.yaml > ibm-block-csi-operator.yaml
+```
+2. Update the image fields in the ibm-block-csi-operator.yaml if the default values are not fit for you.
```
Comment (rivkap) [inline README.md-review]: (Optional): If required, update the image fields in the ibm-block-csi-operator.yaml.

File: README.md-review [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+3. Install the operator
```
Comment (rivkap) [inline README.md-review]: Install the operator.

File: README.md-review [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+<!-- $ kubectl apply -f csi_driver.yaml  (download and install csi_driver.yaml only if you are using Kubernetes v.1.14+) -->
+```bash
+$ kubectl apply -f ibm-block-csi-operator.yaml
+```
+
+### Verify operator is running:
```
Comment (rivkap) [inline README.md-review]: Verify the operator is running:

File: README.md-review [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+```bash
+$ kubectl get pod -l app.kubernetes.io/name=ibm-block-csi-operator -n kube-system
+NAME                                    READY   STATUS    RESTARTS   AGE
+ibm-block-csi-operator-5bb7996b86-xntss 2/2     Running   0          10m
+```
+
+### Create an IBMBlockCSI custom resource
+1. Create an IBMBlockCSI yaml file (ibc.yaml) as following and update the repository and tag if the default values are not fit for you:
```
Comment (rivkap) [inline README.md-review]: 1. Create an IBMBlockCSI yaml file (ibc.yaml). If required, update the repository and tag values.

File: README.md-review [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+spec:
+  controller:
+    repository: stg-artifactory.haifa.ibm.com:5030/ibm-block-csi-controller-driver
+    tag: "1.0.0_b40_origin.develop"
+  node:
+    repository: stg-artifactory.haifa.ibm.com:5030/ibm-block-csi-node-driver
+    tag: "1.0.0_b40_origin.develop"
+```
+
+2. Apply it:
```
Comment (rivkap) [inline README.md-review]: 2. Apply the ibc.yaml file.

File: README.md-review [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+```bash
+$ kubectl apply -f ibc.yaml
+```
+
+### Verify driver is running:
```
Comment (rivkap) [inline README.md-review]: Verify the driver is running:

File: README.md-review [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+ibm-block-csi-controller-0   5/5     Running   0          10m
+
+$ kubectl get -n kube-system pod --selector=app=ibm-block-csi-node
+NAME                       READY   STATUS    RESTARTS   AGE
+ibm-block-csi-node-xnfgp   3/3     Running   0          10m
+ibm-block-csi-node-zgh5h   3/3     Running   0          10m
+
+```
+
+> **Note**: For further usage, please go to https://github.com/IBM/ibm-block-csi-driver
```
Comment (rivkap) [inline README.md-review]: > **Note**: For further usage details, refer to https://github.com/IBM/ibm-block-csi-driver.

File: README.md-review [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+
+## Uninstallation
```
Comment (rivkap) [inline README.md-review]: Uninstalling

File: README.md-review [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+### 1. Delete the IBMBlockCSI custom resource
```
Comment (rivkap) [inline README.md-review]: custom resource**.**

File: README.md-review [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+```bash
+$ kubectl delete -f ibc.yaml
+```
+
+### 2. Delete the operator
```
Comment (rivkap) [inline README.md-review]: Delete the operator. (add a period at the end of the sentence)

File: deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review [hunk 15]
```diff
@@ -0,0 +1,493 @@
...
+          "metadata": {
+            "name": "node-name"
+          },
+          "spec": {}
+        }
+      ]
+spec:
+  displayName: "Operator for IBM block storage CSI driver"
+  description: |
+    **IBM Block CSI driver** is a Container Storage Interface (CSI) Driver for IBM block storage systems which enables container orchestrators to manage the life-cycle of persistent storage.
```
Comment (rivkap) [inline deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review]: **IBM block storage CSI driver** is a Container.....

to manage the life cycle of persistent storage. 
(no hyphen for "life cycle")

File: deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review [hunk 16]
```diff
@@ -0,0 +1,493 @@
...
+
+    This is the official operator to deploy and manage IBM block storage CSI driver.
+
+    Supported container platforms:
+      - Openshift v4.1
```
Comment (rivkap) [inline deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review]: OpenShift

File: deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review [hunk 17]
```diff
@@ -0,0 +1,493 @@
...
+      - Kubernetes v1.13
+
+    Supported IBM storage systems:
+      - IBM FlashSystem 9100
+      - IBM Spectrum Virtualize
+      - IBM Storwize
+      - IBM FlashSystem A9000\R
```
Comment (rivkap) [inline deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review]: IBM FlashSystem A9000/R (make the slash go the other way)

File: deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review [hunk 18]
```diff
@@ -0,0 +1,493 @@
...
+
+    Supported operating systems:
+      - RHEL 7.x (x86 architecture)
+      - RHCOS 4.1 (x86 architecture)
+
+    ## Pre-requisites
```
Comment (rivkap) [inline deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review]: Prerequisites

File: deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review [hunk 19]
```diff
@@ -0,0 +1,493 @@
...
+    
+    ### Worker nodes preparation
+    Perform these steps for each worker node in OpenShift:
+    
+    ### 1. Install Linux packages to ensure Fibre Channel and iSCSI connectivity
```
Comment (rivkap) [inline deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review]: connectivity. 
(add a period at the end of the sentence)

File: deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review [hunk 20]
```diff
@@ -0,0 +1,493 @@
...
+    Skip this step, if the packages are already installed.
```
Comment (rivkap) [inline deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review]: If the packages are already installed, you can skip this step.

File: deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review [hunk 21]
```diff
@@ -0,0 +1,493 @@
...
+    
+    RHEL 7.x:
+    ```bash
+    $ yum -y install sg3_utils
```
Comment (rivkap) [inline deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review]: the sg3_utils was removed from the driver file. please consult with Ran regarding if it should also be removed from here.

File: deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review [hunk 22]
```diff
@@ -0,0 +1,493 @@
...
+    $ multipath -ll
+    ```
+    
+    Important: When configuring Linux multipath devices, verify that the `find_multipaths` parameter in the `multipath.conf` file is disabled.
+      - RHEL 7.x: Remove the `find_multipaths yes` string from the `multipath.conf` file.
+    
+    #### 3. Configure storage system connectivity.
+    3.1. Define the hostname of each OpenShift worker node on the relevant storage systems with the valid WWPN or IQN of the node.
+    
+    3.2. For Fiber Chanel, configure the relevant zoning from the storage to the host.
```
Comment (rivkap) [inline deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review]: Fibre Channel

File: deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review [hunk 23]
```diff
@@ -0,0 +1,493 @@
...
+    3.3.2. Discover and log into at least two iSCSI targets on the relevant storage
+    systems.
+    
+    ```bash
+    $ iscsiadm -m discoverydb -t st -p ${storage system iSCSI port IP}:3260
+    --discover
+    $ iscsiadm -m node -p ${storage system iSCSI port IP/hostname} --login
+    ```
+    
+    3.3.3. Verify that the login was successful and display all targets that you logged in. The portal value must be the iSCSI target IP address.
```
Comment (rivkap) [inline deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review]: Verify that the login was successful and display all targets that you logged into. The portal value must be the iSCSI target IP address.

File: deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review [hunk 24]
```diff
@@ -0,0 +1,493 @@
...
+    
+    ```bash
+    $ iscsiadm -m session --rescan
+    Rescanning session [sid: 1, target: {storage system IQN},
+    portal: {storage system iSCSI port IP},{port number}
+    ```
+
+    ### Install CSIDriver CRD
```
Comment (rivkap) [inline deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review]: In the CSI driver - this is marked as optional 
is it also optional here?

File: deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review [hunk 25]
```diff
@@ -0,0 +1,493 @@
...
+    Enabling CSIDriver on OpenShift
+    1. Enable Technology Preview features using feature gates, more details, please refer to https://docs.openshift.com/container-platform/4.1/nodes/clusters/nodes-cluster-enabling-features.html
```
Comment (rivkap) [inline deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review]: 1. Enable Technology Preview features using feature gates. For more details, refer to https://docs.openshift.com/container-platform/4.1/nodes/clusters/nodes-cluster-enabling-features.html.

File: deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review (hunk 21 above)
Comment (27149chen) [reply deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review]: remove it.

File: deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review (hunk 24 above)
Comment (27149chen) [reply deploy/olm-catalog/ibm-block-csi-operator/1.0.0/ibm-block-csi-operator.v1.0.0.clusterserviceversion.yaml-review]: it is required in OpenShift 4.1

**Comment by oriyarde** [comment]
@27149chen could we delete this branch?

**Pull request merged**